src/subdomains/core/payment-link/dto/payment-link.dto.ts

```diff
@@ -173,7 +173,7 @@ export class PaymentLinkDto {
   lnurl: string;
 
   @ApiPropertyOptional({ type: PaymentLinkPaymentDto })
-  payment?: PaymentLinkPaymentDto;
+  payment?: PaymentLinkPaymentDto | PaymentLinkPaymentDto[];
 
```

Comment:
Might be a bit confusing to have both types in one DTO. Maybe create a separate DTO for the history and use a base class for the common fields?